ipn/prefs.go

```diff
@@ -477,6 +482,92 @@ func (p *Prefs) SetAdvertiseExitNode(runExit bool) {
 		netaddr.IPPrefixFrom(netaddr.IPv6Unspecified(), 0))
 }
 
+// peerWithTailscaleIP returns the peer in st with the provided
+// Tailscale IP.
+func peerWithTailscaleIP(st *ipnstate.Status, ip netaddr.IP) (ps *ipnstate.PeerStatus, ok bool) {
+	for _, ps := range st.Peer {
+		for _, ip2 := range ps.TailscaleIPs {
+			if ip == ip2 {
+				return ps, true
+			}
+		}
+	}
+	return nil, false
+}
+
+func checkIPNotSelf(st *ipnstate.Status, ip netaddr.IP) bool {
+	for _, tip := range st.TailscaleIPs {
+		if ip == tip {
+			return false
+		}
+	}
+	return true
+}
+
+// ClearExitNodeIP sets the ExitNodeIP to the zero value.
```

Comment:
clarify that this doesn't mutate the ExitNodeID, though?

Reply:
Actually we probably want to mutate both (the purpose is for clearing out any persisted exit node info). Updated.